lib/bolt/bolt_option_parser.rb

```diff
@@ -761,6 +761,9 @@ def initialize(options)
       end
 
       separator "\nDISPLAY OPTIONS"
+      define('--filter FILTER', 'Filter tasks and plans by a matching substring') do |filter|
```

Comment:
We should validate that the filter only contains legal task/plan characters. Something like `[a-z0-9_:]`, either `filter =~ /[a-z0-9_:]+/` or `filter !~ /^[a-z0-9_:]/`.